pkg/logql/log/ip.go

```diff
@@ -0,0 +1,195 @@
+package log
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+// This actually implements the Stage interface to process the line
+type LineIPFiler struct {
+	pat string
+}
+
+func NewLineIpFilter(pattern string) (Stage, error) {
+	return &LineIPFiler{pat: pattern}, nil
+}
+
+func (f *LineIPFiler) Process(line []byte, _ *LabelsBuilder) ([]byte, bool) {
+	match, _ := ip(f.pat, string(line))
```

Comment:
The parsing of the AST parameter (e.g ip("127.0.0.1")) should not be for every line but instead only once.

This should also be returned as an AST error if the format is invalid.